modules/tensor_mechanics/src/kernels/InertialForce.C

```diff
@@ -95,25 +98,27 @@ InertialForce::InertialForce(const InputParameters & parameters)
                "integrator.");
 }
 
-Real
-InertialForce::computeQpResidual()
+template <bool is_ad>
+GenericReal<is_ad>
+InertialForceTempl<is_ad>::computeQpResidual()
 {
   if (_dt == 0)
     return 0;
   else if (_has_beta)
   {
-    Real accel = 1. / _beta *
+    auto accel = 1.0 / _beta *
                  (((_u[_qp] - (*_u_old)[_qp]) / (_dt * _dt)) - (*_vel_old)[_qp] / _dt -
                   (*_accel_old)[_qp] * (0.5 - _beta));
-    Real vel = (*_vel_old)[_qp] + (_dt * (1 - _gamma)) * (*_accel_old)[_qp] + _gamma * _dt * accel;
+    auto vel =
+        (*_vel_old)[_qp] + (_dt * (1.0 - _gamma)) * (*_accel_old)[_qp] + _gamma * _dt * accel;
     return _test[_i][_qp] * _density[_qp] *
-           (accel + vel * _eta[_qp] * (1 + _alpha) - _alpha * _eta[_qp] * (*_vel_old)[_qp]);
+           (accel + vel * _eta[_qp] * (1.0 + _alpha) - _alpha * _eta[_qp] * (*_vel_old)[_qp]);
   }
 
   // Lumped mass option
   // Only lumping the masses here
   // will multiply by corresponding residual multiplier after lumping the matrix
-  else if (_time_integrator.isLumped() && _time_integrator.isExplicit())
+  else if (_time_integrator.isLumped() && _time_integrator.isExplicit() && !is_ad)
```

Comment:
Why the additional condition here? (`!is_ad`)

Reply:
Because I haven't figured out how to implement the lumping below in AD.

What's the equivalent of:
```
    _u_dotdot_factor = &_var.vectorTagValue(_time_integrator.uDotDotFactorTag());
```

I didn't see an AD interface for that. @lindsayad

Reply:
I've looked at this `uDotDotFactorTag` relatively recently but it was obfuscated and confusing enough that I've forgotten what it represents. Can you remind me? I need to understand what it is in order to understand whether AD makes sense for it

Reply:
I don't think I get this. Aren't we already using the same tagging for AD and non-AD in other instances of the same file-- with the same interface?

```  /**
   * Returns the tag for the nodal multiplication factor for the residual calculation of the udotdot
   * term.
   *
   * By default, this tag will be associated with udotdot.
   */
  TagID uDotDotFactorTag() const { return _u_dotdot_factor_tag; }

Reply:
This was introduced in https://github.com/idaholab/moose/pull/14658/commits/bfbdc37ad8fa357654217b40a014262e84725d30

Reply:
Anyway if this is coming from a tagged (residual) vector, then there's no hope of getting AD information out of it. To support something like that you would need to introduce `DistributedVectors` of `AD` type into the tagging system